fl4health/strategies/feddg_ga_with_adaptive_constraint.py

```
@@ -0,0 +1,268 @@
+            evaluate_metrics_aggregation_fn : Optional[MetricsAggregationFn]
+                Metrics aggregation function, optional.
+            adapt_loss_weight (bool, optional): Determines whether the value of mu is adaptively modified by
+                the server based on aggregated train loss. Defaults to False.
+            loss_weight_delta (float, optional): This is the amount by which the server changes the value of mu
+                based on the modification criteria. Only applicable if adaptivity is on. Defaults to 0.1.
+            loss_weight_patience (int, optional): This is the number of rounds a server must see decreasing
+                aggregated train loss before reducing the value of mu. Only applicable if adaptivity is on.
+                Defaults to 5.
+            weighted_train_losses (bool, optional): Determines whether the training losses from the clients should be
+                aggregated using a weighted or unweighted average. These aggregated losses are used to adjust the
+                proximal weight in the adaptive setting. Defaults to False.
+            fairness_metric : FairnessMetric, optional.
+                The metric to evaluate the local model of each client against the global model in order to
+                determine their adjustment weight for aggregation. Can be set to any default metric in
+                FairnessMetricType or set to use a custom metric. Optional, default is
+                FairnessMetric(FairnessMetricType.LOSS).
+            weight_step_size : float
+                The step size to determine the magnitude of change for the adjustment weight. It has to be
+                0 < weight_step_size < 1. Optional, default is 0.2.
+        """
+        if fraction_fit != 1.0 or fraction_evaluate != 1.0:
+            log(
+                WARNING,
+                "fraction_fit or fraction_evaluate are not 1.0. The behaviour of FedDG-GA is unknown in those cases.",
```

Should we just throw an AssertionError if the behaviour is undefined? Or is there scenarios where we want to this anyway

---

The method can technically work in these cases. However, the behavior isn't considered in the original paper and there are a few edge case type things that aren't well specified. I think your suggestion is a good one. Let's throw an error so the user doesn't do anything they really don't want to do.